### PR TITLE
Initial Immutant support

### PR DESCRIPTION
**Pull request by tobias**
This adds an Immutant adapter (#13), along with tests. Note that the tests won't pass until #14 is merged.

### CODE REVIEW

File: src/spiral/adapters/immutant.clj [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                             (-> (response "nil response body in spiral.core/to-immutant")
+                               (status 500))
+                             :close? true)))
+              error-chan ([e]
+                          (clojure.stacktrace/print-cause-trace e)
+                          (immutant/send! ch
+                            (-> (response (str "Encountered error! See log."))
+                              (status 500))
+                            :close? true)
+                          (log/error e)))))))))
```
Comment (dgrnbrg) [inline src/spiral/adapters/immutant.clj]: Could you use the form `(log/error e "Encountered unhandled error from spiral error channel")`? If you don't pass an error string, some version of clojure.tools.logging don't print the stacktrace then.

File: src/spiral/adapters/immutant.clj [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+              resp-chan ([resp]
+                         (if resp
+                           (immutant/send! ch resp
+                             :close? (not (:websocket? req)))
+                           (immutant/send! ch
+                             (-> (response "nil response body in spiral.core/to-immutant")
+                               (status 500))
+                             :close? true)))
+              error-chan ([e]
+                          (clojure.stacktrace/print-cause-trace e)
```
Comment (dgrnbrg) [inline src/spiral/adapters/immutant.clj]: I'm unsure whether we should ever write to stdout/stderr; I feel like only sending to the logging lib is sufficient--what do you think?

Comment (tobias) [inline src/spiral/adapters/immutant.clj]: Yep, I just used the http-kit adapter as a template for this one. I can leave this if you like, but will add a string to the `log/error` call below.

**Comment by dgrnbrg** [comment]
(Close & reopen triggers a rebuild, based on the recently merged dependent PR)

**Comment by dgrnbrg** [comment]
Thank you for adding Immutant support!

**Comment by tobias** [comment]
My pleasure! I'd be happy to provide a PR for updating the README as well, if you like.

**Comment by dgrnbrg** [comment]
That would be great. I started going through to do that myself, but I would
trust your code examples to be more idiomatic in their using of immutant
namespaces :)

**Pull request merged**